R/fill.R

```diff
@@ -19,8 +19,91 @@ NULL
 #'   or "updown" (first up and then down).
 #' @export
 #' @examples
-#' df <- data.frame(Month = 1:12, Year = c(2000, rep(NA, 11)))
-#' df %>% fill(Year)
+#' # Value (year) is recorded only when it changes
+#' sales <- tibble::tribble(
+#'   ~quarter, ~year, ~sales,
+#'   "Q1",  2000,  66013,
+#'   "Q2",    NA,  69182,
+#'   "Q3",    NA,  53175,
+#'   "Q4",    NA,  21001,
+#'   "Q1",  2001,  46036,
+#'   "Q2",    NA,  58842,
+#'   "Q3",    NA,  44568,
+#'   "Q4",    NA,  50197,
+#'   "Q1",  2002,  39113,
+#'   "Q2",    NA,  41668,
+#'   "Q3",    NA,  30144,
+#'   "Q4",    NA,  52897,
+#'   "Q1",  2004,  32129,
+#'   "Q2",    NA,  67686,
+#'   "Q3",    NA,  31768,
+#'   "Q4",    NA,  49094
+#' )
+#'
+#' # `fill()` will default to replacing missing data from top to bottom
+#' sales %>% fill(year)
+#'
+#'
+#' # Value (pet_type) is missing above
+#' tidy_pets <- tibble::tribble(
+#'   ~rank, ~pet_type,                          ~breed,
```

Comment:
`~breed` needs to come to the left a little. And I think maybe you could reduce it to 6 breeds per type?